hazelcast/test/src/HazelcastTests7.cpp

```diff
@@ -1322,12 +1322,8 @@ namespace hazelcast {
                     clientConfig.get_network_config().get_aws_config().set_enabled(true).
                             set_access_key(std::getenv("AWS_ACCESS_KEY_ID")).set_secret_key(std::getenv("AWS_SECRET_ACCESS_KEY")).
                             set_tag_key("aws-test-tag").set_tag_value("aws-tag-value-1");
-
-#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
```

Comment:
if you delete tis line, how will we test `.set_inside_aws(true)` API? This code runs at the windows aws machine.

Reply:
I think there should be separate test cases (they can be parametrized) that use `.set_inside_aws(true)` and they should only be run if the corresponding runner is on AWS.

Reply:
yes, it can be done, lets put these in place before removing these usages so that we do not lose the coverage.

Reply:
I have parametrized the first 3 tests of the `AwsClientTest` suite. They will run both with `set_inside_aws(true)` and `set_inside_aws(false)`. The `true` one will be skipped if the environment variable `INSIDE_AWS` is not set. I will add `set INSIDE_AWS=1` for Windows jobs on Jenkins.